Asset resource docs (#521)
mulesoft/docs-runtime-manager: modules/ROOT/pages/anypoint-platform-cli-commands.adoc

|<<exchange asset deprecate>> | Deprecates an asset
|<<exchange asset undeprecate>> | Undeprecate an asset
|<<exchange asset describe>> | Show a given asset's information

> |<<exchange asset resource upload>> | Upload a resource to an asset page

Reviewer: ```suggestion
|<<exchange asset resource upload>> | Upload a resource to an asset portal
```

> |<<exchange asset resource upload>> | Upload a resource to an asset page

Author: ```suggestion
|<<exchange asset resource upload>> | Upload a resource to an asset portal
```

> |<<exchange asset resource upload>> | Upload a resource to an asset portal

Reviewer: ```suggestion
|<<exchange asset resource upload>> | Uploads a resource to an asset portal
```

|<<exchange asset resource list>> | Lists published resources in the asset portal
|<<exchange asset resource download>> | Downloads resource from the asset portal

> |<<exchange asset resource delete>> | Delete resource from the asset portal as draft and then publish that draft

Reviewer: ```suggestion
|<<exchange asset resource delete>> | Deletes resource from the asset portal as draft and then publishes that draft
```
Adding final 's' to be consistent.

Reviewer: I'm not sure I understand this sentence. I would rather have the following and maybe explain it in the command's section 
```suggestion
|<<exchange asset resource delete>> | Delete resource from the asset portal
```

Naming the page "home" makes the uploaded page the main description page for the Exchange asset.

[NOTE]
This command will publish any active draft as part of the operation.

> Argument `assetIdentifier` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +

Reviewer 2: Why is `group_id` identified with `(` and `[`? 
Can this be formatted `<group_id>/<asset_id>/<version>` as shown in the [exchange asset upload](https://docs.mulesoft.com/runtime-manager/anypoint-platform-cli-commands#exchange-asset-upload) command?

I see other published instances where we document:
> ([groupID]/)assetID/version

Is there a difference between them?

Author: It's an optional argument. 

```
Argument "assetIdentifier" should be formatted as follows: ([group_id]/)<asset_id>/<version>. If "group_id" is not specified, it will default to the currently selected Organization ID.
```

Reviewer 2: Thank you. 
In that case, I think it would make more sense to go with 
> `[group_id]/<asset_id>/<version>`

At least from the docs we never followed a convention about how to denote optional attributes. But overall it tends to be accepted that `[]` denotes "optional", while `<>` denotes "required".

This command describes the asset passed in `<assetIdentifier>`.

Argument `assetIdentifier` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +
If `group_id` is not specified, it defaults to the currently selected Organization ID.

// CLI DEFAULTS

> include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

Reviewer: 👀 
![image](https://user-images.githubusercontent.com/62706868/125303296-14e6a900-e303-11eb-8e74-99565e216526.png)

Author: 🤷‍♂️ all doc content has the same link to default optional arguments

=== exchange asset resource upload

----
> exchange asset resource upload [options] <assetIdentifier> <filepath>
----

> This command uploads a resource `<filepath>` to an asset page describes in `<assetIdentifier>`.

Reviewer: ```suggestion
This command uploads a resource `<filepath>` to an asset page described in `<assetIdentifier>`.
```

Argument `assetIdentifier` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +
If `group_id` is not specified, it defaults to the currently selected Organization ID.

You can use `--draft` option to list draft resources in the asset portal.

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== exchange asset resource download

----
> exchange asset resource download [options] <assetIdentifier> <resourcePath> <filePath>
----

This command downloads the specified published resource `<resourcePath>` from the asset portal `<assetIdentifier>` to `<filePath>`

Argument `<assetIdentifier>` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +
If `<group_id>` is not specified, it defaults to the currently selected Organization ID.

Argument `<resourcePath>` should be a resource path of a publicated resource. You can list publicated resources with `list` command (exchange asset resource list)

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== exchange asset resource delete

----
> exchange asset resource delete [options] <assetIdentifier> <resourcePath>
----

> This command delete `<resourcePath>` from the asset `<assetIdentifier>` in asset portal as draft and then publish that draft

Reviewer: ```suggestion
This command deletes the resource `<resourcePath>` from the asset `<assetIdentifier>` by publishing a new draft  in which the resource has been deleted. 
```

Maybe something like that.

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== exchange asset resource upload

----
> exchange asset resource upload [options] <assetIdentifier> <filepath>
----

> This command uploads a resource `<filepath>` to an asset page described in `<assetIdentifier>`.

Reviewer 3: ```suggestion
This command uploads a resource from `<filepath>` to an asset portal described in `<assetIdentifier>`.
```

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== exchange asset resource upload

----
> exchange asset resource upload [options] <assetIdentifier> <filepath>
----

> This command uploads a resource from `<filepath>` to an asset portal described in `<assetIdentifier>`.
> 
> Argument `assetIdentifier` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +
> If `group_id` is not specified, it defaults to the currently selected Organization ID.
> 
> Supported file extensions: `jpeg`, `jpg`, `jpe`, `gif`, `bmp`, `png`, `webp`, `ico`, `svg`, `tiff`, `tif`.
> 
> The successful output command will be a markdown codesnippet.
> You can use it on any page of your `<assetIdentifier>` asset.

Reviewer 2: ```suggestion
This command uploads the resource specified in `<filepath>` to a page in the asset portal described in `<assetIdentifier>`.

You can use this command for any page of your `<assetIdentifier>` asset.

Supported file extensions for `<filepath>` are: `jpeg`, `jpg`, `jpe`, `gif`, `bmp`, `png`, `webp`, `ico`, `svg`, `tiff`, `tif`.

The argument `assetIdentifier` should be formatted as follows: `[group_id]/<asset_id>/<version>`. +
If `group_id` is not specified, it defaults to the currently selected Organization ID.

The successful output command will be a markdown codesnippet.
```

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== exchange asset resource list

----
> exchange asset resource list [options] <assetIdentifier>
----

> This command lists published resources for `<assetIdentifier>` in the asset portal
> 
> Argument `assetIdentifier` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +
> If `group_id` is not specified, it defaults to the currently selected Organization ID.
> 
> [NOTE]
> This command lists published resources by default. +
> You can use the `--draft` option to list draft resources from the asset portal.
> 
> // CLI DEFAULTS
> include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

Reviewer 2: ```suggestion
This command lists the resources in the asset portal of the asset specified in `<assetIdentifier>`.

Argument `assetIdentifier` should be formatted as follows: `[group_id]/<asset_id>/<version>`. +
If `group_id` is not specified, it defaults to the currently selected Organization ID.

[NOTE]
This command lists published resources by default. +
You can use the `--draft` option to list draft resources from the asset portal.

Besides the default `--help`, `-f`/`--fields` and `-o`/`--output` options, this command also takes the `--draft` option to list non-published resources in the asset portal.
```

=== exchange asset resource download

----
> exchange asset resource download [options] <assetIdentifier> <resourcePath> <filePath>
----

> This command downloads the specified published resource `<resourcePath>` from the asset portal `<assetIdentifier>` to `<filePath>`
> 
> Argument `<assetIdentifier>` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +
> If `<group_id>` is not specified, it defaults to the currently selected Organization ID.
> 
> Argument `<resourcePath>` should be a resource path of a publicated resource. You can list publicated resources with `list` command (`exchange asset resource list`)

Reviewer 2: ```suggestion
This command downloads the published resource specified in `<resourcePath>` from the asset portal of the asset specified in `<assetIdentifier>` to the file specified in `<filePath>`.

Argument `assetIdentifier` should be formatted as follows: `[group_id]/<asset_id>/<version>`. +
If `group_id` is not specified, it defaults to the currently selected Organization ID.

Argument `<resourcePath>` must be a published resource. +
You can list all published resources using the <<exchange asset resource list,asset resource list>> command.
```

[NOTE]
This command only supports published resources.

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== exchange asset resource delete

----
> exchange asset resource delete [options] <assetIdentifier> <resourcePath>
----

> This command deletes the resource `<resourcePath>` from the asset `<assetIdentifier>` by publishing a new draft portal where the resource has been deleted.
> 
> Argument `<assetIdentifier>` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +
> If `<group_id>` is not specified, it defaults to the currently selected Organization ID.
> 
> Argument `<resourcePath>` should be a resource path of a publicated resource. You can list publicated resources with `list` command (`exchange asset resource list`)
> 
> [NOTE]
> This command will publish any active draft as part of the operation.

Reviewer 2: ```suggestion
This command deletes the resource specified in `<resourcePath>` from the asset portal of the asset specified in `<assetIdentifier>` by publishing a new portal in which `resourcePath` has been deleted.

Argument `<assetIdentifier>` should be formatted as follows: `[group_id]/<asset_id>/<version>`. +
If `group_id` is not specified, it defaults to the currently selected Organization ID.

Argument `<resourcePath>` must be a published resource. +
You can list all published resources using the <<exchange asset resource list,asset resource list>> command.

[CAUTION]
This command publishes all active drafts as part of the operation.

```

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== exchange asset resource upload

----
> exchange asset resource upload [options] <assetIdentifier> <filepath>
----

This command uploads a resource `<filepath>` to an asset page described in `<assetIdentifier>`.

Argument `assetIdentifier` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +
If `group_id` is not specified, it defaults to the currently selected Organization ID.

> Supported file extensions: jpeg, jpg, jpe, gif, bmp, png, webp, ico, svg, tiff, tif.

Reviewer 4: maybe adding "`" between all the supported extensions?

The successful output command will be a markdown codesnippet.
You can use it on any page of your `<assetIdentifier>` asset.

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== exchange asset resource list

----
> exchange asset resource list [options] <assetIdentifier>
----

This command lists published resources for `<assetIdentifier>` in the asset portal

Argument `assetIdentifier` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +
If `group_id` is not specified, it defaults to the currently selected Organization ID.

> [NOTE]
> This command only supports published resources.
> 
> You can use `--draft` option to list draft resources in the asset portal.

Reviewer 2: Do we have other _types_ (?) here besides "published" or "draft"? 
Because it's easier to read everything in the same note, and we should change the wording a little bit:

```suggestion
[NOTE]
This command lists published resources by default. +
You can use the `--draft` option to list draft resources from the asset portal.
```

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== exchange asset resource download

----
> exchange asset resource download [options] <assetIdentifier> <resourcePath> <filePath>
----

This command downloads the specified published resource `<resourcePath>` from the asset portal `<assetIdentifier>` to `<filePath>`

Argument `<assetIdentifier>` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +
If `<group_id>` is not specified, it defaults to the currently selected Organization ID.

> Argument `<resourcePath>` should be a resource path of a publicated resource. You can list publicated resources with `list` command (exchange asset resource list)

Reviewer 4: ```suggestion
Argument `<resourcePath>` should be a resource path of a publicated resource. You can list publicated resources with `list` command (`exchange asset resource list`)
```

[NOTE]
This command only supports published resources.

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== exchange asset resource delete

----
> exchange asset resource delete [options] <assetIdentifier> <resourcePath>
----

> This command deletes the resource `<resourcePath>` from the asset `<assetIdentifier>` by publishing a new draft  in which the resource has been deleted. 

Reviewer 4: ```suggestion
This command deletes the resource `<resourcePath>` from the asset `<assetIdentifier>` by publishing a new draft portal where the resource has been deleted. 
```

Argument `<assetIdentifier>` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +
If `<group_id>` is not specified, it defaults to the currently selected Organization ID.

> Argument `<resourcePath>` should be a resource path of a publicated resource. You can list publicated resources with `list` command (exchange asset resource list)

Reviewer 4: ```suggestion
Argument `<resourcePath>` should be a resource path of a publicated resource. You can list publicated resources with `list` command (`exchange asset resource list`)
```
